invenio_records_resources/services/records/service.py

```diff
@@ -198,7 +209,8 @@ def _create(self, record_cls, identity, data, links_config=None):
             self,
             identity,
             record,
-            links_config=links_config
+            links_config=links_config,
+            errors=errors
```

Comment:
comment: Have you discussed how the UI will show errors/warnings if they are coming from saving a draft? I understand that in this case the save will be complete but still, errors will be returned! Will we have a different way of showing these warnings? If that's the case, why don't we differentiate them already here in "warnings" or something that indicates that these errors at that point are not fatal for the UI...

Reply:
I think the overall idea is that for updates (i.e. `PUT /api/records/:id/draft`), we basically always give back a 200, with an `errors` key. For publishing though (i.e. `POST /api/records/:id/draft/actions/publish`), we give back a 400 in case of validation errors... The question is, should we keep the response payload homogenous, i.e. still return both the partial valid record, and the `errors` key at the top level?

Reply:
> ...we basically always give back a 200, with an `errors` key (if there are errors)...

There are some "fancier" ways of communicating a partial success, via the [`207 Multi-Status` response](https://evertpot.com/http/207-multi-status), but since there's nothing "standard", probably let's stick with what we know :)

Reply:
yeah I was thinking of sticking to 200 / 400 and then like @slint mentioned: because we know the context in which we are getting a payload back with errors (following a save draft or a publish), the frontend can use that to tweak the display. Save draft would display with `warning` (orange) and publish with `error` (red).